pkg/cluster/internal/delete/delete.go

```diff
@@ -35,15 +35,16 @@ func Cluster(logger log.Logger, p providers.Provider, name, explicitKubeconfigPa
 
 	kerr := kubeconfig.Remove(name, explicitKubeconfigPath)
 	if kerr != nil {
-		logger.Errorf("failed to update kubeconfig: %v", kerr)
+		return kerr
+	} else {
+		logger.V(0).Infof("Removed context %q from kubeconfig", kubeconfig.ContextForCluster(name))
```

Comment:
this isn't strictly true, the context may not have been present in the cluster

this probably is most easily accomplished inside Remove

Reply:
Remove just returns RemoveKIND, which doesn't seem to have access to the logger.

Reply:
@BenTheElder what do you think about modifying RemoveKIND() to add the logger as a parameter?

Reply:
Ping @BenTheElder

Reply:
alternatively we could return true/false if the cluster is removed or not.

I think it would be reasonable to add the logger as a parameter.